addons/product/models/product_pricelist.py

```diff
@@ -345,6 +345,15 @@ def get_import_templates(self):
             'label': _('Import Template for Pricelists'),
             'template': '/product/static/xls/product_pricelist.xls'
         }]
+    
+    def unlink(self):
+        for pricelist in self:
+            linked_items = self.env['product.pricelist.item'].sudo().with_context(active_test=False).search(
+                [('base', '=', 'pricelist'), ('base_pricelist_id', '=', pricelist.id)])
```

Comment:
```suggestion
                [('base', '=', 'pricelist'), ('base_pricelist_id', '=', pricelist.id), ('pricelist_id', 'not in', self.ids)])
```
Hopefully last remark :D.  If we delete a bunch of pricelists together (or all the pricelists), we should not force the user to delete them separately because they are linked together ;).